### DISCUSSION TITLE
Livemap does not seem to work work Nucleus

### DISCUSSION THREAD

**Discussion by kloentje2** [TGRHavoc/live_map#103] [General]
Hello,

I am working on the livemap and want to use it in my server. I already have the web repo installed on my nginx server and have found the resource for the FiveM server. This is the config: 

```
set socket_port 30121
set livemap_debug "warn" # "[all]" 'trace', 'debug', 'info', 'warn', 'error', 'fatal', 'off'
set blip_file "server/blips.json"
set livemap_access_control "*"
set livemap_use_nucleus true # Allow livemap to set up a secure reverseProxy using the Nucleus project
```

When I go to IP:30121/blips.json I do see the blips, but I can't use that (ofc.) because the IP is not secured by HTTPS/TLS/SSL

I am using Nucleus and I get the connection JSON

"reverseProxy": {"socket":"wss://monitor-***.users.cfx.re","blips":"https://monitor-***.users.cfx.re/blips"} (apparently not equal to the username of the server owner?!) 

But when I do this I can't connect to the websocket it does not work
![0Schermafbeelding 2024-08-25 143410](https://github.com/user-attachments/assets/395c1d61-e5b8-4eba-b163-aecf3728bcd0)

Can you please help me?

**Comment by TGRHavoc** [comment]
Yeah, there's a fundamental incompatibility between the current implementation of the resource and nucleus. Currently I'm binding the socket to its own port which, doesn't get routed through nucleus.

I am working on a v4 (read: trying to find time to work on) which, uses the SetHttpHandler for implementing server sent events which, will automatically get routed through nucleus.

If you want to use the current version on a SSL secured site, you will need to set up a reverse proxy.
